src/util/ChartUtils.ts

```diff
@@ -573,27 +573,44 @@ export const getCoordinatesOfGrid = (ticks: Array<TickItem>, min: number, max: n
  * @param {Boolean} isAll Return the ticks of all the points or not
  * @return {Array}  Ticks
  */
-export const getTicksOfAxis = (axis: any, isGrid?: boolean, isAll?: boolean): TickItem[] => {
+export const getTicksOfAxis = (
+  axis: BaseAxisProps & {
+    duplicateDomain?: any;
+    realScaleType?: 'scaleBand' | 'band' | 'point' | 'linear';
+    scale?: any;
+    axisType?: AxisType;
+    ticks?: any;
+    niceTicks?: any;
+    isCategorical?: boolean;
+    categoricalDomain?: any;
+  },
+  isGrid?: boolean,
+  isAll?: boolean,
+): TickItem[] | null => {
   if (!axis) return null;
   const { scale } = axis;
   const { duplicateDomain, type, range } = axis;
 
   const offsetForBand = axis.realScaleType === 'scaleBand' ? scale.bandwidth() / 2 : 2;
   let offset = (isGrid || isAll) && type === 'category' && scale.bandwidth ? scale.bandwidth() / offsetForBand : 0;
 
-  offset = axis.axisType === 'angleAxis' ? mathSign(range[0] - range[1]) * 2 * offset : offset;
+  offset = axis.axisType === 'angleAxis' && range?.length >= 2 ? mathSign(range[0] - range[1]) * 2 * offset : offset;
 
-  // The ticks setted by user should only affect the ticks adjacent to axis line
+  // The ticks set by user should only affect the ticks adjacent to axis line
   if (isGrid && (axis.ticks || axis.niceTicks)) {
-    return (axis.ticks || axis.niceTicks).map((entry: TickItem) => {
+    const result = (axis.ticks || axis.niceTicks).map((entry: TickItem) => {
       const scaleContent = duplicateDomain ? duplicateDomain.indexOf(entry) : entry;
 
       return {
+        // If the scaleContent is not a number, the coordinate will be NaN.
+        // That could be the case for example with a PointScale and a string as domain.
         coordinate: scale(scaleContent) + offset,
         value: entry,
         offset,
       };
     });
+
+    return result.filter((row: TickItem) => !_.isNaN(row.coordinate));
```

Comment:
Finally the fix.

Reply:
So there is no way to fix this upstream just because of the way that `PointScale` works?

Reply:
Upstream nothing is broken. This is about the domain being filtered by the brush - as it should - and the PointScale simply has no coordinates for items outside of the domain. 

That is how the point scale works.

Reply:
To my best understanding, all works as it should - but this function.

Reply:
Got it. Logically in my head I think I'd want the data itself to be filtered and then everything trickle down into domain and ticks from there. Makes sense, but these are the kind of fragile pieces we have to keep an eye on and harden/test

Reply:
Also do we want to use `_.isNaN(row.coordinate)` as opposed to `Number.isNaN(row.coordinate)`? The more we use lodash the more we have dependency on it (I know its used a ton but just future thinking)